idaes/apps/grid_integration/prescient_plugin.py

```diff
@@ -0,0 +1,524 @@
+prescient.plugins.add_custom_commandline_option(opt)
+
+### End add new command line options ###
+
+from strategic_bidding import DAM_thermal_bidding
+from tracking import DAM_thermal_tracking
+
+def initialize_customized_results(options, simulator):
+
+    simulator.data_manager.extensions['customized_results'] = {}
+    customized_results = simulator.data_manager.extensions['customized_results']
+
+    customized_results['Generator'] = []
+    customized_results['Date'] = []
+    customized_results['Hour'] = []
+    customized_results['State'] = []
+    customized_results['RUC Schedule'] = []
+    customized_results['SCED Schedule'] = []
+    customized_results['Power Output'] = []
+
+    return
+prescient.plugins.register_initialization_callback(initialize_customized_results)
+
+def initialize_bidding_object(options, simulator):
+
+    '''
+    This function initializes the bidding object and stores it in the Prescient
+    data manager.
+
+    Arguments:
+        options: Prescient options.
+        simulator: Prescient simulator.
+    Returns:
+        None
+    '''
+
+    # initialize the model class
+    bidder = DAM_thermal_bidding(rts_gmlc_data_dir = options.rts_gmlc_data_dir, \
+                                 price_forecast_file = options.price_forecast_file,\
+                                 generators = [options.bidding_generator],\
+                                 horizon = options.ruc_horizon)
+
+    plugin_dict = simulator.data_manager.extensions.setdefault('plugin_objects',{})
+    plugin_dict['bidder'] = bidder
+
+    return
+prescient.plugins.register_initialization_callback(initialize_bidding_object)
+
+def initialize_tracking_object(options, simulator):
+
+    '''
+    This function initializes the tracking object and stores it in the Prescient
+    data manager.
+
+    Arguments:
+        options: Prescient options.
+        simulator: Prescient simulator.
+    Returns:
+        None
+    '''
+
+    # initialize the model class
+    tracker = DAM_thermal_tracking(rts_gmlc_data_dir = options.rts_gmlc_data_dir,\
+                                   tracking_horizon = options.sced_horizon,\
+                                   generators = [options.bidding_generator])
+
+    plugin_dict = simulator.data_manager.extensions.setdefault('plugin_objects',{})
+    plugin_dict['tracker'] = tracker
+
+    return
+prescient.plugins.register_initialization_callback(initialize_tracking_object)
+
+def pass_DA_bid_to_prescient(options, ruc_instance, bids):
+
+    '''
+    This method passes the bids into the RUC model for day-ahead market clearing.
+
+    Arguments:
+        options: Prescient options.
+        ruc_instance: Prescient RUC object
+        bids: the bids we want to pass into the day-ahead market. It is a
+        dictionary that has this structure {t: {generator: {power: cost}}}.
+    Returns:
+        None
+    '''
+
+    gen_name = options.bidding_generator
```

Comment:
Think about the case where we replace a node in RTS-GMLC with a full IDAES IES model. We might not use the RTS-GMLC data to define the bidding and tracker. Right now, your bidder and tracker uses the RTS-GMLC data directly.